update.go

```diff
@@ -331,42 +331,73 @@ func initUpdate() error {
 			return
 		}
 
-		doneCh := make(chan struct{})
-		go func() {
-			killSupervisedServices()
-			close(doneCh)
-		}()
-
-		atLeast1Second := time.After(time.Second)
-
-		go func() {
-			select {
-			case <-doneCh:
-				log.Println("All processes shut down")
-			case <-time.After(15 * time.Second):
-				log.Println("Timed out waiting for processes to shut down")
+		signalDelay := time.Minute
+		if s := r.FormValue("wait_per_signal"); s != "" {
+			var err error
+			signalDelay, err = time.ParseDuration(s)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
 			}
+		}
 
-			// give the HTTP response some time to be sent
-			<-atLeast1Second
+		rc := http.NewResponseController(w)
+		start := time.Now()
+		killSupervisedServicesAndUmountPerm(signalDelay)
+		fmt.Fprintf(w, "All services killed in %s\n", time.Since(start))
+		rc.Flush()
 
-			log.Println("Unmounting /perm")
-			if err := syscall.Unmount("/perm", unix.MNT_FORCE); err != nil {
-				log.Printf("unmounting /perm failed: %v", err)
-			}
+		log.Println("Rebooting")
+		if err := reboot(); err != nil {
```

Comment:
Can you keep the reboot() call in a separate goroutine here? The current code breaks the gokrazy updater, because it doesn’t complete the HTTP request before rebooting.